### PR TITLE
Feat/oddysey events

### LINKED ISSUE #28: [Missing Event] CreateSuitLoadout

### LINKED ISSUE #29: [Missing Event] DeleteSuitLoadout

### LINKED ISSUE #33: [Missing Event] Embark

### LINKED ISSUE #34: [Missing Event] LoadoutEquipModule

### LINKED ISSUE #35: [Missing Event] LoadoutRemoveModule

### LINKED ISSUE #36: [Missing Event] RenameSuitLoadout

### LINKED ISSUE #37: [Missing Event] ScanOrganic

### LINKED ISSUE #38: [Missing Event] SellMicroResources

### LINKED ISSUE #39: [Missing Event] SellOrganicData

### LINKED ISSUE #40: [Missing Event] SellSuit

### LINKED ISSUE #41: [Missing Event] SellWeapon

### LINKED ISSUE #42: [Missing Event] ShipLockerMaterials

### LINKED ISSUE #43: [Missing Event] SwitchSuitLoadout

### LINKED ISSUE #44: [Missing Event] TransferMicroResources

### LINKED ISSUE #45: [Missing Event] TradeMicroResources

### LINKED ISSUE #46: [Missing Event] UpgradeSuit

### LINKED ISSUE #47: [Missing Event] UpgradeWeapon

### LINKED ISSUE #48: [Missing Event] UseConsumable

### PR DESCRIPTION
**Pull request by Pjiesco**
Added missing events

- UseConsumable
  - closes #48
- UpgradeWeapon  
  - closes #47
- UpgradeSuit
  - closes #46  
- TradeMicroResources
  - closes #45
- TransferMicroResources
  - closes #44
- SwitchSuitLoadout
  - closes #43
- ShipLockerMaterials
  - closes #42
- SellWeapon
  - closes #41
- SellSuit 
  - closes #40
- SellOrganicData
  - closes #39 
- SellMicroResources
  - closes #38
- ScanOrganic
  - closes #37
- RenameSuitLoadout
  - closes #36
- LoadoutRemoveModule
  - closes #35
- LoadoutEquipModule
  - closes #34 
- Embark
  - closes #33
- DeleteSuitLoadout
  - closes #29 
- CreateSuitLoadout
  - closes #28

### CODE REVIEW

File: src/main/java/com/christophecvb/elitedangerous/events/backpack/TransferMicroResourcesEvent.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.christophecvb.elitedangerous.events.Event;
+import com.christophecvb.elitedangerous.models.MicroResource;
+
+import java.util.ArrayList;
+
+public class TransferMicroResourcesEvent extends Event {
+    public ArrayList<transfer> transfers;
+
+    public static class transfer extends MicroResource {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/backpack/TransferMicroResourcesEvent.java]: rename `transfer` to `MicroResourceTransfer`

File: src/main/java/com/christophecvb/elitedangerous/events/backpack/TransferMicroResourcesEvent.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.christophecvb.elitedangerous.events.backpack;
+
+import com.christophecvb.elitedangerous.events.Event;
+import com.christophecvb.elitedangerous.models.MicroResource;
+
+import java.util.ArrayList;
+
+public class TransferMicroResourcesEvent extends Event {
+    public ArrayList<transfer> transfers;
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/backpack/TransferMicroResourcesEvent.java]: change `ArrayList` to `List`

File: src/main/java/com/christophecvb/elitedangerous/events/buysell/SellMicroResourcesEvent.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.christophecvb.elitedangerous.events.buysell;
+
+import com.christophecvb.elitedangerous.events.Event;
+import com.christophecvb.elitedangerous.models.MicroResource;
+
+import java.util.ArrayList;
+
+public class SellMicroResourcesEvent extends Event {
+    public ArrayList<MicroResource> microResources;
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/buysell/SellMicroResourcesEvent.java]: `ArrayList` -> `List`

File: src/main/java/com/christophecvb/elitedangerous/events/buysell/SellOrganicDataEvent.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import com.christophecvb.elitedangerous.events.Event;
+import com.christophecvb.elitedangerous.models.OrganicData;
+
+import java.util.ArrayList;
+
+public class SellOrganicDataEvent extends Event {
+    public Integer marketID;
+    public ArrayList<BioData> bioData;
+
+    public static class BioData extends OrganicData {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/buysell/SellOrganicDataEvent.java]: Why does BioData and OrganicData are 2 separate classes?

File: src/main/java/com/christophecvb/elitedangerous/events/buysell/SellSuitEvent.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.christophecvb.elitedangerous.events.buysell;
+
+import com.christophecvb.elitedangerous.events.Event;
+
+public class SellSuitEvent extends Event {
+    public String name;
+    public String nameLocalised;
+    public Long price;
+}
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/buysell/SellSuitEvent.java:10]: Missing SuitID

File: src/main/java/com/christophecvb/elitedangerous/events/loadout/LoadoutBaseEvent.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.christophecvb.elitedangerous.events.loadout;
+
+import com.christophecvb.elitedangerous.events.Event;
+
+public abstract class LoadoutBaseEvent extends Event {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/loadout/LoadoutBaseEvent.java]: rename `LoadoutBaseEvent` to `LoadoutEvent`

File: src/main/java/com/christophecvb/elitedangerous/events/loadout/SwitchSuitLoadoutEvent.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.christophecvb.elitedangerous.events.loadout;
+
+import java.util.ArrayList;
+
+public class SwitchSuitLoadoutEvent extends LoadoutBaseEvent {
+    public ArrayList<Module> modules;
+
+    public static class Module {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/loadout/SwitchSuitLoadoutEvent.java]: Rename that `LoadoutModule` and move it to models package

File: src/main/java/com/christophecvb/elitedangerous/events/loadout/LoadoutRemoveModuleEvent.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.christophecvb.elitedangerous.events.loadout;
+
+public class LoadoutRemoveModuleEvent extends LoadoutEquipModuleEvent {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/loadout/LoadoutRemoveModuleEvent.java]: Can't do that. What you can do is create a abstract class `LoadoutModuleEvent extends LoadoutEvent` and then have the `Equip` and `Remove` extending that one

File: src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package com.christophecvb.elitedangerous.events.other;
+
+import com.christophecvb.elitedangerous.events.Event;
+
+public class EmbarkEvent extends Event {
+    public Boolean SRV;
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java]:
```
@SerializedName("SRV")
public Boolean srv;
```

File: src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    public Boolean taxi;
+    public Boolean multicrew;
+    public Long ID;
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java]:
```
@SerializedName("ID")
public Boolean io;
```

File: src/main/java/com/christophecvb/elitedangerous/events/trade/TradeMicroResourcesEvent.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import com.christophecvb.elitedangerous.models.MicroResource;
+
+import java.util.ArrayList;
+
+public class TradeMicroResourcesEvent extends Event {
+    public ArrayList<MicroResource> offered;
+    public String received;
+    public Integer count;
+    public Long marketID;
+}
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/trade/TradeMicroResourcesEvent.java:14]: Missing `Category`
https://elite-journal.readthedocs.io/en/latest/New%20in%20Odyssey/#trademicroresources

File: src/main/java/com/christophecvb/elitedangerous/models/OrganicData.java [hunk 12]
```diff
@@ -0,0 +1,6 @@
+package com.christophecvb.elitedangerous.models;
+
+public class OrganicData {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/models/OrganicData.java]: Same question, how OrganicData is different from BioData?

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]
Almost first try! Great job ;)

File: src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java (hunk 10 above)
Comment (Pjiesco) [reply src/main/java/com/christophecvb/elitedangerous/events/other/EmbarkEvent.java]: io or id?

File: src/main/java/com/christophecvb/elitedangerous/events/loadout/CreateSuitLoadoutEvent.java [hunk 13]
```diff
@@ -1,7 +1,9 @@
 package com.christophecvb.elitedangerous.events.loadout;
 
+import com.christophecvb.elitedangerous.models.Module;
+
 import java.util.ArrayList;
 
-public class CreateSuitLoadoutEvent extends LoadoutBaseEvent {
-    public ArrayList<SwitchSuitLoadoutEvent.Module> modules;
+public class CreateSuitLoadoutEvent extends LoadoutEvent {
+    public ArrayList<Module> modules;
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/loadout/CreateSuitLoadoutEvent.java]: `Module` -> `LoadoutModule`

File: src/main/java/com/christophecvb/elitedangerous/events/buysell/SellOrganicDataEvent.java [hunk 14]
```diff
@@ -1,15 +1,16 @@
...
-import com.christophecvb.elitedangerous.models.OrganicData;
 
 import java.util.ArrayList;
 
 public class SellOrganicDataEvent extends Event {
     public Integer marketID;
     public ArrayList<BioData> bioData;
 
-    public static class BioData extends OrganicData {
+    public static class BioData {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/events/buysell/SellOrganicDataEvent.java]: Rename to `BioDataItem` ?

File: src/main/java/com/christophecvb/elitedangerous/models/LoadoutModule.java [hunk 15]
```diff
@@ -0,0 +1,7 @@
+package com.christophecvb.elitedangerous.models;
+
+public class LoadoutModule {
```
Comment (ChristopheCVB) [inline src/main/java/com/christophecvb/elitedangerous/models/LoadoutModule.java:3]: Use it now?!

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]

File: src/main/java/com/christophecvb/elitedangerous/events/loadout/CreateSuitLoadoutEvent.java (hunk 13 above)
Comment (ChristopheCVB) [reply src/main/java/com/christophecvb/elitedangerous/events/loadout/CreateSuitLoadoutEvent.java]: Do not rename it, just use the correct Class

**Review by ChristopheCVB — DISMISSED** [review]

**Review by ChristopheCVB — APPROVED** [review]

**Pull request merged**